### ISSUE TITLE
Keepass password input bug?

### ISSUE THREAD

**Issue by zb140** [twpayne/chezmoi#768]
## Describe the bug

Not sure if this is just on Windows, but I ran across a problem while trying to put together a secret caching script with KeepassXC as suggested in #763 .  When I try to use the `keepassxc` template function, I get this error:
```
Insert password to unlock c:\users\zb\.op_cache\cache.db:
chezmoi: template: C:\Users\zb\.local\share\chezmoi\1pass-test.tmpl:1:13: executing "C:\\Users\\zb\\.local\\share\\chezmoi\\1pass-test.tmpl" at <keepassxc "foo/bar">: error calling keepassxc: keepassxc: keepassxc-cli.exe show --show-protected 'c:\\users\\zb\\.op_cache\\cache.db' foo/bar: read stdin: The handle is invalid.
```
I believe it's due to this line: https://github.com/twpayne/chezmoi/blame/master/cmd/secretkeepassxc.go#L133 . It looks like it's trying to read the password from stdout.  Indeed, changing it to `os.Stdin.Fd()` fixes the error and I'm able to enter the password and read secrets from my db successfully.

## To reproduce

1. Create a keepassxc database with some stuff in it
1. Attempt to read it with the `keepassxc` template function

## Output of command with `--verbose`

```
$ chezmoi command --verbose
```
The error is pasted above.

## Output of `chezmoi doctor`

```
$ chezmoi doctor
     ok: version 1.8.1, commit 4def4f3b96cb01cefe8d7887e8dbfb77908ef267, built at 2020-05-06T18:05:06Z, built by goreleaser
     ok: runtime.GOOS windows, runtime.GOARCH amd64
  ERROR: C:\Users\zb\.local\share\chezmoi (source directory, perm 777)
     ok: C:\Users\zb (destination directory, perm 777)
     ok: C:\Users\zb\.config\chezmoi\chezmoi.toml (configuration file)
     ok: C:\WINDOWS\system32\cmd.exe (shell)
     ok: c:\users\zb\.op_cache\cache.db (KeePassXC database)
     ok: C:\WINDOWS\gvim.bat (editor)
     ok: C:\WINDOWS\vimdiff.bat (merge command)
     ok: C:\Program Files\Git\cmd\git.exe (source VCS command, version 2.26.2)
     ok: C:\Program Files\Git\usr\bin\gpg.exe (GnuPG, version 2.2.20)
     ok: C:\Users\zb\scoop\shims\op.exe (1Password CLI, version 1.0.0)
warning: bw (Bitwarden CLI, not found)
     ok: C:\Users\zb\scoop\shims\gopass.exe (gopass CLI, version 1.9.2)
     ok: C:\Program Files\KeePassXC\keepassxc-cli.exe (KeePassXC CLI, version 2.5.4)
warning: lpass (LastPass CLI, not found)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

## Additional context

Like I said, I'm not sure if this is just a Windows problem -- I haven't tried it on Linux yet, but it seems like others would have encountered this if it were a more widespread issue.

Also, ultimately I'd like to be able to pipe the password in (or provide it some way other than interactively), but I've just tried that and it doesn't work either with `os.Stdout` or with my updated `os.Stdin` version. No idea why yet...

**Comment by twpayne** [comment]
Oops, that looks like a clear bug in chezmoi. It's possible that you're the first person to use KeePassXC support. I added support for a bunch of password managers some time, but they're not well tested (I hadn't discovered `testscript` at the time). 

A PR to fix this would be very welcome, even better if it included an integration test in the style of `testdata/scripts/secretlastpass.txt`.

**Comment by zb140** [comment]
Ok, I've got a fix ready, and I added the integration test too. Implementing the test actually uncovered the reason why I couldn't pipe a password in: `terminal.ReadPassword` can only read passwords from an actual terminal (which makes some sense in hindsight). While investigating, I came across this issue already filed upstream that goes into lots more detail: https://github.com/golang/go/issues/19909

This was causing my piping issue, but also prevented the test script from running correctly as the call to `chezmoi apply` is not attached to a terminal.  To fix this, I used one of the wrapper implementations in the upstream issue's comments that can read from either a terminal or a non-terminal stdin. Let me know what you think.
